### PR TITLE
Ability to set the ID when creating Clusters and Tenants

### PR DESCRIPTION
**Pull request by corvus-ch**
Implementes #98.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update the ./CHANGELOG.md.
- [x] Link this PR to related issues.

### CODE REVIEW

File: pkg/service/cluster_test.go [hunk 1]
```diff
@@ -88,12 +88,54 @@ func TestCreateCluster(t *testing.T) {
...
+	request api.Id
+	response api.Id
+}{
+	"requested ID gets accepted": {
+		"c-my-custom-id",
+		"c-my-custom-id",
+	},
+	"ID without prefix gets prefixed": {
+		"my-custom-id",
+		"c-my-custom-id",
```
Comment (ccremer) [inline pkg/service/cluster_test.go:101]: I'd suggest to set the properties explicitly
```suggestion
		request: "c-my-custom-id",
		response: "c-my-custom-id",
	},
	"ID without prefix gets prefixed": {
		request: "my-custom-id",
		response: "c-my-custom-id",
```

File: pkg/service/tenant_test.go [hunk 2]
```diff
@@ -75,6 +75,49 @@ func TestCreateTenant(t *testing.T) {
...
+	request api.Id
+	response api.Id
+}{
+	"requested ID gets accepted": {
+		"t-my-custom-id",
+		"t-my-custom-id",
+	},
+	"ID without prefix gets prefixed": {
+		"my-custom-id",
+		"t-my-custom-id",
```
Comment (ccremer) [inline pkg/service/tenant_test.go:88]: same here

File: pkg/service/tenant.go [hunk 3]
```diff
@@ -42,15 +43,19 @@ func (s *APIImpl) CreateTenant(c echo.Context) error {
 		*newTenant.GitRepo.Url == "" {
 		return echo.NewHTTPError(http.StatusBadRequest, "GitRepo URL is required")
 	}
-	apiTenant := &api.Tenant{
-		TenantProperties: api.TenantProperties(*newTenant),
+	apiTenant := api.Tenant(*newTenant)
+	if !strings.HasPrefix(string(apiTenant.Id), api.TenantIDPrefix) {
```
Comment (ccremer) [inline pkg/service/tenant.go:47]: I'm seeing this `string(id)` more often now, which kind of annoys me a little. I think it makes sense to implement the `Stringer` interface for the id like
```go
func (in Id) String() string {
  return string(in)
}
```
```suggestion
	if !strings.HasPrefix(apiTenant.Id.String()), api.TenantIDPrefix) {
```
Shouldn't bee too difficult.
You could also do that in a separate PR if you would like to separate this change.

Comment (corvus-ch) [reply pkg/service/tenant.go:47]: Ah, got point. Totally missed that one.

**Review by srueg — APPROVED** [review]

**Comment by srueg** [comment]
Please test this implementation on the SYNfra cluster in the `lieutenant-dev` namespace before merging.
You can either build & deploy a test image or run the operator locally against the cluster.

**Review by ccremer — APPROVED** [review]
